commons/src/main/java/org/infinispan/commons/collections/SynchronizedMiniList.java

```diff
@@ -0,0 +1,126 @@
+package org.infinispan.commons.collections;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+/**
+ * Synchronized version of {@link MiniList}
+ *
+ * Note that iterators are not synchronized ({@link java.util.Collections#synchronizedList()} does not provide that neither).
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class SynchronizedMiniList<T> extends MiniList<T> {
```

Comment:
Seems my reply got lost... Collections.synchronizedList allocates additional object wrapping the original one, while SynchronizedMiniList just inherits - it's still just one object.